clang/test/CXX/drs/dr1807.cpp

```diff
@@ -0,0 +1,31 @@
+// RUN: %clang_cc1 -std=c++98 %s -triple x86_64-linux-gnu -emit-llvm -o - -fexceptions -fcxx-exceptions -pedantic-errors | llvm-cxxfilt -n | FileCheck %s --check-prefixes CHECK,CXX98
+// RUN: %clang_cc1 -std=c++11 %s -triple x86_64-linux-gnu -emit-llvm -o - -fexceptions -fcxx-exceptions -pedantic-errors | llvm-cxxfilt -n | FileCheck %s --check-prefixes CHECK,SINCE-CXX11
+// RUN: %clang_cc1 -std=c++14 %s -triple x86_64-linux-gnu -emit-llvm -o - -fexceptions -fcxx-exceptions -pedantic-errors | llvm-cxxfilt -n | FileCheck %s --check-prefixes CHECK,SINCE-CXX11
+// RUN: %clang_cc1 -std=c++17 %s -triple x86_64-linux-gnu -emit-llvm -o - -fexceptions -fcxx-exceptions -pedantic-errors | llvm-cxxfilt -n | FileCheck %s --check-prefixes CHECK,SINCE-CXX11
+// RUN: %clang_cc1 -std=c++20 %s -triple x86_64-linux-gnu -emit-llvm -o - -fexceptions -fcxx-exceptions -pedantic-errors | llvm-cxxfilt -n | FileCheck %s --check-prefixes CHECK,SINCE-CXX11
+// RUN: %clang_cc1 -std=c++23 %s -triple x86_64-linux-gnu -emit-llvm -o - -fexceptions -fcxx-exceptions -pedantic-errors | llvm-cxxfilt -n | FileCheck %s --check-prefixes CHECK,SINCE-CXX11
+// RUN: %clang_cc1 -std=c++2c %s -triple x86_64-linux-gnu -emit-llvm -o - -fexceptions -fcxx-exceptions -pedantic-errors | llvm-cxxfilt -n | FileCheck %s --check-prefixes CHECK,SINCE-CXX11
+
+namespace dr1807 { // dr1807: 3.0
+struct S {
+  S() {}
+  ~S() {}
+};
+
+void f() {
+  try {
+    S s[3];
+  } catch (...) {}
+}
+}
+
+// CHECK:              invoke void @dr1807::S::S(){{.+}}
+// CHECK-NEXT:         {{.+}} unwind label %lpad
+
+// CHECK-LABEL:      lpad:
```

Comment:
these labels aren't stable/some modes of the build bots remove them, so you'll have to use a wildcard for these.

Reply:
> some modes of the build bots remove them

How unfortunate. How do they look instead? Numbers?

Reply:
Yeah, they look like numbers.  Also interesting, parameter names disappear completely, and  %0, %1, %2 (up to # params) refer to them withoutnaming them.

Reply:
This test is using clang_cc1 with a fixed target. I don't see how different output is possible here.

Reply:
> This test is using clang_cc1 with a fixed target. I don't see how different output is possible here.

See `-fdiscard-value-names`, which is pretty common on as a default on some downstreams, resource sensitive build bots, and even some folk's local builds.

Reply:
The clang driver may or may not pass `-discard-value-names` to cc1 (depending on the build type), but for cc1 the default is always to not discard value names. This (and similar issues) is exactly why cc1 is always used for codegen tests.

Reply:
My takeaway here is that we don't need to worry about names being discarded, but what about stability of names in general? Names I'm hard-coding in this particular test have been stable for 12 years (since 3.0), but I'm not sure if I can generalize this.

Reply:
> The clang driver may or may not pass `-discard-value-names` to cc1 (depending on the build type), but for cc1 the default is always to not discard value names. This (and similar issues) is exactly why cc1 is always used for codegen tests.

My understanding is it is a build option for the compiler.  I've DEFINITELY gotten build bot complaints for leaving names in in the past.